x-pack/test/security_solution_api_integration/test_suites/detections_response/detection_engine/rule_execution_logic/frozen_indices_handling/trial_license_complete_tier/frozen_indices_handling.ts

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import expect from '@kbn/expect';
+import { CreateRuleProps } from '../../../../../../../api_integration/services/security_solution_api.gen';
+
+import { dataGeneratorFactory } from '../../../../utils';
+import {
+  deleteAllRules,
+  deleteAllAlerts,
+} from '../../../../../../../common/utils/security_solution';
+import { FtrProviderContext } from '../../../../../../ftr_provider_context';
+
+export default ({ getService }: FtrProviderContext) => {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const log = getService('log');
+  const securitySolutionApi = getService('securitySolutionApi');
+  const retry = getService('retry');
+
+  describe('@ess frozen indices queried', () => {
+    const index = 'test_idx';
+    const { indexListOfDocuments } = dataGeneratorFactory({ es, index, log });
+
+    const cleanOperationsLIFO: Array<() => Promise<void>> = [];
+    const addCleanOperation = (operation: () => Promise<void>) =>
+      cleanOperationsLIFO.push(operation);
+    const runCleanOperations = async () => {
+      for (let idx = cleanOperationsLIFO.length - 1; idx >= 0; idx--) {
+        await cleanOperationsLIFO[idx]();
+      }
+    };
+
+    after(async () => {
+      await deleteAllAlerts(supertest, log, es);
+      await deleteAllRules(supertest, log);
+      await runCleanOperations();
+    });
+
+    it('should report the querying of frozen indices during rule execution', async () => {
```

Comment:
Let's split this test into two tests where one checks the non-frozen test data case and one checks with frozen data. I think it'd also be useful to extract the frozen data creation logic into a separate function and run that in `before()`, then have the individual tests contain just the rule execution and rule result fetching logic.

Reply:
As discussed, I have split the tests into 3 cases:
- Querying a regular index that is not frozen
- Querying an index that is frozen
- Querying an index that is not frozen, but has a name that starts with `partial-`